app/src/main/java/com/github/factotum_sdp/factotum/data/localisation/Location.kt

```diff
@@ -53,45 +54,54 @@ class Location(query: String, context : Context) {
             if(!alreadyExists) cacheFile.appendText(toStore)
             return location
         }
+
+        fun geocoderQuery(query: String, context: Context): List<Address>?{
+            // must handle differently depending on SDK.
+            // getLocationFromName(String, int) deprecated in SDK 33
+            val geocoder = Geocoder(context)
+            return if (Build.VERSION.SDK_INT >= TIRAMISU) {
+                tiramisuResultHandler(query, geocoder)
+            } else{
+                resultHandler(query, geocoder)
+            }
+        }
+
+        @RequiresApi(TIRAMISU)
+        private fun tiramisuResultHandler(query: String, geocoder: Geocoder): List<Address>? {
+            var result : List<Address>? = listOf()
+            val latch = CountDownLatch(1)
+            // blocking
+            geocoder.getFromLocationName(query, MAX_RESULT) { addresses ->
+                result = if(addresses.size > 0) addresses else null
+                latch.countDown()
+            }
+            latch.await()
+            return result
+        }
+
+        private fun resultHandler(query: String, geocoder: Geocoder): List<Address>? {
+            var result : List<Address>? = listOf()
+            try {
+                val geocodeResult = geocoder.getFromLocationName(query, MAX_RESULT)
```

Comment:
So getFromLocationName is apparently deprecated, no alternative function or way of doing this?

Reply:
The use of the deprecated function is necessary because of the apk used by the CI. The CI is using APK 30, and other functions force APK 33 to be used.